packages/core/src/shared/plugins/event-editor/useFocusEditorEvents.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+
+import type { Value } from '@udecode/plate-common';
+
+import type { PlateEditor } from '../../types';
+
+import { BLUR_EDITOR_EVENT, FOCUS_EDITOR_EVENT } from './constants';
+
+export const useFocusEditorEvents = ({
```

Comment:
What's the advantage of using custom DOM events for this rather than the existing `useFocused` hook and `useEffect`?

Example:

```ts
const focused = useFocused();
const isFirst = useRef(false);

useEffect(() => {
  if (isFirst.current) {
    isFirst.current = true;
    return;
  }

  if (focused) {
    alert('Editor became focused');
  } else {
    alert('Editor became unfocused');
  }
}, [focused]);
```